### PR TITLE
bodyscanners, autodocs and sleepers now eject their occupants properly when destroyed

### LINKED ISSUE #3368: Autodoc yeets people into oblivion when destroyed
## Testmerges:

Server-Side Code Modifications: https://tgstation13.org/parsed-logs/basil/server_side_modifications.dm

Round ID: 1894
BYOND Version: 512.1488
Server revision compiled on: 
Master commit: 6215a912803ea6de76a8c14fd9ec657dab2c6d25
The following pull requests are currently test merged:
#3365: 'Headless reviving fix' by Rohesie at commit d272c34f2b
#3309: 'Unga juice nerf and chem additions' by jroinc at commit 8a004ee9ad
#3359: 'Feedback fixes' by Rohesie at commit 9f7f293c81
#3358: 'New attack animations' by Rohesie at commit 830a069366
#3347: 'Xeno ERT fix' by Rohesie at commit 30bf90503c
#3343: 'Legacy chat support re-enabled' by Rohesie at commit cf2744d5e1
#3335: 'Fixes Carrier being able to drop 10 huggers in a group of marines in about a second' by terranaut1 at commit 2ddd76d46c
#3282: 'AFK, Respawn and Game tab tweaks' by Rohesie at commit 35dbc16c2d
#3331: 'Shotgun spread tweaks' by Rohesie at commit 79e232ada1
#3327: 'Keybind tweaks' by Rohesie at commit 5eef94e7f7
#3350: 'Better disk spawns' by BurgerLUA at commit 3310fff6f2
#3305: 'Converts marine gloves to greyscale' by Mark9013100 at commit e31450e24c
#3357: 'Adds a New Map: Desert Outpost AKA "Dessert"' by BurgerLUA at commit 2dfd3028d7
#3363: 'm81 grenade launcher fix' by terranaut1 at commit 2a7537e0cb

Local commit: 00f2f76880168279a1d7509d6f587f3304fbbc8a
Server tools version: /tg/station 13 Server v3.2.6.2

## Reproduction:

1. Get inside an autodoc
2. Get it destroyed by either making a crusher ram on it, acid it and etc
3. Watch as your character goes to the shadow wealm

### PR DESCRIPTION
**Pull request by terranaut1**
## About The Pull Request

Fixes #3368 

adds existing go_out or similar procs to the machines' destroy()
adds a forceeject var and related code to the autodoc and adds it to its destroy()

## Why It's Good For The Game

no more yeetus deleetus

## Changelog
:cl:
fix: Fixes Autodoc sending you into the Shadowrealm when destroyed
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
This is a good general solution.

**Comment by terranaut1** [comment]
I'll leave it as-is unless there's a reason machines shouldn't dump their contents. Also saves me the trouble of putting it in the bodyscanner and sleeper too (whose shared subtype is machinery) and avoids unexpected behavior with any future machines that might hold mobs or items.

**Comment by Rohesie** [comment]
I'll wait for Spooky's feedback.

File: code/game/objects/machinery/autodoc.dm [hunk 1]
```diff
@@ -674,6 +676,14 @@
 		playsound(loc,'sound/machines/buzz-two.ogg', 25, 1)
 		return
 	if(occupant)
+		if(forceeject)
+			if(!surgery)
+				visible_message("\The [src] is destroyed, ejecting [occupant] and showering them in debris.")
+				occupant.take_limb_damage(rand(10,20),rand(10,20))
+				go_out(AUTODOC_NOTICE_FORCE_EJECT)
+			visible_message("\The [src] malfunctions as it is destroyed mid-surgery, ejecting [occupant] with surgical wounds and showering them in debris.")
```
Comment (Rohesie) [inline code/game/objects/machinery/autodoc.dm]: Does a return fit here? Else there will be two messages and two `go_out` being called.

Comment (terranaut1) [reply code/game/objects/machinery/autodoc.dm]: It didn't cause gameplay issues but a runtime where it tried to damage null.limb
Oops

**Pull request merged**